### DISCUSSION TITLE
wazuh send alerts to gmail

### DISCUSSION THREAD

**Discussion by sinungp** [Security-Onion-Solutions/securityonion#6662] [Unsupported Versions]
Hello

I want to ask how to configure wazuh so that it can send alerts to gmail?

thank you

**Comment by dougburks** [comment]
Wazuh does not support SMTP authentication so it can't send directly to Gmail. You can build a mail relay (outside of Security Onion):
https://documentation.wazuh.com/current/user-manual/manager/manual-email-report/smtp-authentication.html

Then configure Wazuh to send mail through the relay:
https://docs.securityonion.net/en/2.3/email.html#wazuh

**Reply by sinungp** [reply]
Thanks for quick reply

I've made a relay server

when I enter localhost in the smptp server section in /opt/so/conf/wazuh/ossec.conf

 **<smtp_server>localhost</smtp_server>**

out a warning on /nsm/wazuh/logs/ossec.log

like this

**2021/12/24 00:59:41 wazuh-modulesd:download: INFO: Module started.
2021/12/24 00:59:41 wazuh-modulesd:control: INFO: Starting control thread.
2021/12/24 00:59:44 ossec-authd: INFO: Started (pid: 608).
2021/12/24 00:59:44 ossec-authd: INFO: Accepting connections on port 1515. No password required.
2021/12/24 00:59:44 ossec-authd: INFO: Setting network timeout to 1.000000 sec.
2021/12/24 00:59:44 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/24 00:59:52 ossec-maild: ERROR: (1223): Error Sending email to 127.0.0.1 (smtp server)
2021/12/24 00:59:53 rootcheck: INFO: Ending rootcheck scan.
2021/12/24 00:59:54 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/24 01:00:04 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/24 01:00:06 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/24 01:00:06 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.**

is there an error in my configuration?

thank you

**Reply by dougburks** [reply]
If your relay server is outside of Security Onion, then you would need to change `<smtp_server>localhost</smtp_server>` replacing `localhost` with the hostname or IP address of your actual relay server.

**Reply by sinungp** [reply]
Hello

I have created an smtp server on security onion with ip 10.10.5.22

I've changed the configuration to like this

**<smtp_server>10.10.5.22</smtp_server>**

but i still got error like this
is there anything else i can do?


**2021/12/29 01:49:02 ossec-maild: ERROR: (1223): Error Sending email to 10.10.5.22 (smtp server)
2021/12/29 01:49:03 rootcheck: INFO: Ending rootcheck scan.
2021/12/29 01:49:09 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:10 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:10 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:10 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:10 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:10 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:10 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:11 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:19 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:19 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:19 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:19 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:19 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:19 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:19 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:19 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:19 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:26 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:26 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:29 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.
2021/12/29 01:49:39 ossec-remoted: WARNING: (1213): Message from '172.17.0.1' not allowed. Cannot find the ID of the agent. Source agent ID is unknown.**


thanks very much

**Reply by dougburks** [reply]
Did you set all of the `email` options as shown at https://docs.securityonion.net/en/2.3/email.html#wazuh?

**Reply by sinungp** [reply]
I have followed the format

here is the configuration i have

     <email_notification>yes</email_notification>
     <smtp_server>10.10.5.22</smtp_server>
     <email_from>sender@gmail.com</email_from>
     <email_to>reciver.com</email_to>
     <email_maxperhour>12</email_maxperhour>

Thank you

**Reply by dougburks** [reply]
Have you tried manually connecting to port 25 on the SMTP server?

**Reply by sinungp** [reply]
Hello

Sorry, I do not understand what you mean

can you give me a more detailed explanation?

thank you

**Reply by dougburks** [reply]
I'm referring to using something like `telnet` or `nc` to verify that you can manually connect to and send mail through your SMTP server. Troubleshooting SMTP servers is beyond the scope of this discussion forum, but Google should provide several helpful guides.

**Reply by sinungp** [reply]
okay

I have tried to send the message manually with the following command

**echo "This is a test." | mail -s "test message" receiver@gmail.com**

then the message "test message" goes into my gmail

**Reply by dougburks** [reply]
Using the `mail` command may not be exactly the same as connecting to port 25.

**Reply by sinungp** [reply]
Hello

I've tried it using telnet, and it works fine

when I check the logs on /nsm/wazuh/logs/ossec.log

then the error that comes out is like this
2022/01/17 03:28:26 ossec-maild: ERROR: (1223): Error Sending email to 10.10.5.22 (smtp server)
2022/01/17 03:34:32 ossec-maild: ERROR: (1223): Error Sending email to 10.10.5.22 (smtp server)
2022/01/17 03:40:28 ossec-maild: ERROR: (1223): Error Sending email to 10.10.5.22 (smtp server)
2022/01/17 03:58:27 ossec-maild: ERROR: (1223): Error Sending email to 10.10.5.22 (smtp server)
2022/01/17 04:22:27 ossec-maild: ERROR: (1223): Error Sending email to 10.10.5.22 (smtp server)
2022/01/17 04:28:27 ossec-maild: ERROR: (1223): Error Sending email to 10.10.5.22 (smtp server)
2022/01/17 04:46:28 ossec-maild: ERROR: (1223): Error Sending email to 10.10.5.22 (smtp server)
2022/01/17 04:52:28 ossec-maild: ERROR: (1223): Error Sending email to 10.10.5.22 (smtp server)
2022/01/17 05:10:29 ossec-maild: ERROR: (1223): Error Sending email to 10.10.5.22 (smtp server)
2022/01/17 05:16:29 ossec-maild: ERROR: (1223): Error Sending email to 10.10.5.22 (smtp server)

is this error different from the previous one?

thank you

**Reply by dougburks** [reply]
You could try using tcpdump to inspect the SMTP traffic and see if you can determine what the error is, but again troubleshooting SMTP servers is beyond the scope of this discussion forum.
